src/date-picker/date-picker.jsx

```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
   },
 
   propTypes: {
+    /**
+     * Constructor for time formatting.
+     * Follow this specificaction: ECMAScript Internationalization API 1.0 (ECMA-402).
+     */
     DateTimeFormat: React.PropTypes.func,
+
+    /**
+     * If true, automatically accept and close the picker on select a date.
+     */
     autoOk: React.PropTypes.bool,
+
+    /**
+     * Used to control how the DatePicker will be displayed when a user tries to set a date.
+     * `dialog` (default) displays the DatePicker as a dialog with a modal.
+     * `inline` displays the DatePicker below the input field (similar to auto complete)
+     */
     container: React.PropTypes.oneOf(['dialog', 'inline']),
+
+    /**
+     * This is the initial date value of the component.
+     * If either `value` or `valueLink` is provided they will override this
+     * prop with `value` taking precedence.
+     */
     defaultDate: React.PropTypes.object,
+
+    /**
+     * This function is called to format the date to display in the input box.
+     * By default, date objects are formatted to M/D/YYYY.
+     */
     formatDate: React.PropTypes.func,
+
+    /**
+     *
```

Comment:
@oliviertassinari I don't see it being used in Date Picker Dialog. Maybe its been removed?